### PR TITLE
k8s 1.21

### PR DESCRIPTION
**Pull request by kinarashah**
https://github.com/rancher/rancher/issues/31567

### CODE REVIEW

**Review by luthermonson — DISMISSED** [review]

**Comment by StrongMonkey** [comment]
We should probably add https://github.com/rancher/rancher/issues/32180. The node-local-dns manifest looks out-of-date and is causing issues.

File: rke/templates/templates.go [hunk 1]
```diff
@@ -94,7 +97,8 @@ func LoadK8sVersionedTemplates() map[string]map[string]string {
 			">=1.8.0-rancher0 <1.13.0-rancher0":     calicov18,
 		},
 		kdm.Canal: {
-			">=1.20.4-rancher1-1":                      canalv3171,
+			">=1.21.0-rancher1-1":                      canalv319,
+			">=1.20.4-rancher1-1 <1.21.0-rancher1-1":   canalv3171,
```
Comment (StrongMonkey) [inline rke/templates/templates.go:103]: Is there any difference that we treat calico and canal differently?
For >=1.20.4-rancher1-1 <1.21.0-rancher1-1 calico using is v319 but canal is using v3171

Comment (kinarashah) [reply rke/templates/templates.go:103]: Nope, it needs to be `calicov3171` for calico as well, thanks for catching that!

**Comment by kinarashah** [comment]
Will squash commits before merge.

File: rke/templates/templates.go [hunk 2]
```diff
@@ -152,12 +158,14 @@ func LoadK8sVersionedTemplates() map[string]map[string]string {
 			// New ingress template introduced for 1.15.12-rancher1-1, 1.16.10-rancher1-1, 1.17.6-rancher1-1
 			">=1.15.12-rancher1-1 <1.16.1-rancher1-1": nginxIngressV11512,
 			">=1.16.10-rancher1-1 <1.17.0-rancher1-1": nginxIngressV11512,
-			">=1.17.6-rancher1-1":                     nginxIngressV11512,
+			">=1.17.6-rancher1-1 <1.21.0-rancher1-1":  nginxIngressV11512,
+			">1.21.0-rancher1-1":                      nginxIngressv046,
```
Comment (StrongMonkey) [inline rke/templates/templates.go]: THere should be a >= ?

Comment (kinarashah) [reply rke/templates/templates.go]: 😭 updating!

File: rke/k8s_service_options.go [hunk 3]
```diff
@@ -380,6 +388,15 @@ func getKubeAPIOptions120() map[string]string {
 	return kubeAPIOptions
 }
 
+func getKubeAPIOptions121() map[string]string {
+	kubeAPIOptions := getKubeAPIOptions116()
+	// need to turn off this feature in k8s 1.20 since we are not using it https://github.com/kubernetes/kubernetes/pull/91921
```
Comment (StrongMonkey) [inline rke/k8s_service_options.go]: `kubeAPIOptions["feature-gates"] = "ServiceAccountIssuerDiscovery=false"`
Do we need this? I noticed that this is not the exact option compared to 120. Also comment needs to be updated to say 1.21 :)

Comment (kinarashah) [reply rke/k8s_service_options.go]: So ServiceAccountIssuerDiscovery went GA with 1.21 and is now true by default, we no longer need it.
Oh that's right, the comment is no longer required - removing

**Review by StrongMonkey — DISMISSED** [review]
LGTM!

**Review by superseb — DISMISSED** [review]

**Review by StrongMonkey — APPROVED** [review]

**Pull request merged**